### PR TITLE
feat: Unblock user UseCase

### PR DESCRIPTION
**Pull request by borichellow**
[Task-2084](https://wearezeta.atlassian.net/browse/AR-2084)

# What's new in this PR?
- `UnblockUserUseCase` for unblocking blocked users
- UnitTests for it

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/connection/UnblockUserUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+fun interface UnblockUserUseCase {
+    /**
+     * Use case [UnblockUserUseCase] operation
+     *
+     * @param userId the target user whom to unblock
+     * @return a [Either] indicating the operation result:
+     * Left - [CoreFailure] if occurred,
+     * Right - [Unit] is success
+     */
+    suspend operator fun invoke(userId: UserId): Either<CoreFailure, Unit>
```
Comment (yamilmedina) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/connection/UnblockUserUseCase.kt:24]: I believe that this return type was supposed to not leak to clients, if I recall correctly from Prev. Android collectives 🤔  We are wrapping the result in a new type.

**Review by yamilmedina — COMMENTED** [review]
Great work, just a small comment about `Either` leaking into clients, but I would prefer other opinions :D

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/connection/UnblockUserUseCase.kt (hunk 1 above)
Comment (gongracr) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/connection/UnblockUserUseCase.kt:24]: Yepp, it was already agreed that Kalium Use Cases should not expose the `Either` construction to consuming app. The correct way would be to provide a wrapping `Result` class in the same fashion than other use cases do.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/connection/UnblockUserUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import com.wire.kalium.logic.kaliumLogger
+
+/**
+ * Use Case that allows the current self user to unblock another previously blocked user
+ */
+fun interface UnblockUserUseCase {
+    /**
+     * Use case [UnblockUserUseCase] operation
+     *
+     * @param userId the target user whom to block
```
Comment (gongracr) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/connection/UnblockUserUseCase.kt:17]:
```suggestion
     * @param userId the target user whom to unblock
```

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/connection/UnblockUserUseCaseTest.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Test
+    fun givenUnblockingSuccessful_thenBlockResultIsSuccess() = runTest {
+        val (_, blockUser) = Arrangement()
+            .withBlockResult(Either.Right(TestConnection.CONNECTION))
+            .arrange()
+
+        val result = blockUser(TestUser.USER_ID)
+
+        assertTrue(result is UnblockUserResult.Success)
```
Comment (gongracr) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/connection/UnblockUserUseCaseTest.kt:42]: Can we check at least that the `connectionRepository.updateConnectionStatus` was invoked?

**Review by gongracr — APPROVED** [review]
Tiny non blocking comments but looking good otherwise 🕺 🎈

**Pull request merged**